### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by Kangz**

### CODE REVIEW

File: src/pp.rs [hunk 1]
```diff
@@ -104,8 +104,8 @@ pub fn convert_lexer_token(token: LexerToken) -> Result<Token, (PreprocessorErro
             location,
         }),
 
-        LexerTokenValue::NewLine => Err((PreprocessorError::UnexpectedNewLine, location)),
         LexerTokenValue::Hash => Err((PreprocessorError::UnexpectedHash, location)),
+        LexerTokenValue::NewLine => unreachable!(),
```
Comment (JCapucho) [inline src/pp.rs:110]: Could you explain why this is `unreachable`

File: src/pp.rs [hunk 2]
```diff
@@ -778,9 +778,7 @@ impl MacroProcessor {
         loop {
             // Get the next token (without additional expansion)
             let token = match self.step_no_continue(lexer) {
-                Err(StepExit::Continue) => {
-                    continue;
-                }
+                Err(StepExit::Continue) => unreachable!(),
```
Comment (JCapucho) [inline src/pp.rs]: why can't we keep the continue here even if it's unreachable now it might help in the future if we accidentally break this assumption

**Review by JCapucho — APPROVED** [review]
LGTM, but maybe it would be better to add some explanation about the new `unreachable` calls

File: src/pp.rs (hunk 2 above)
Comment (Kangz) [reply src/pp.rs]: Changed to use `step` instead of `step_no_continue` and put the `continue` right here instead of the `unreachable`.

Comment (JCapucho) [reply src/pp.rs]: Oh didn't notice the `step_no_continue` ignore this then

**Pull request merged**